libraries/AP_ESC_Telem/AP_ESC_Telem_HobbyWing_Telem.h

```diff
@@ -0,0 +1,40 @@
+/*
+  ESC Telemetry for HobbyWing devices
+ */
+
+#pragma once
+
+#include "AP_ESC_Telem_config.h"
+
+// AP_ESC_Telem interface
+#if AP_ESC_TELEM_HOBBYWING_ENABLED
+class AP_ESC_Telem_HobbyWing
+    : AP_ESC_Telem_Backend
+{
+public:
+
+    // constructor
+    AP_ESC_Telem_HobbyWing();
+
+    static const struct AP_Param::GroupInfo var_info[];
+
+    void init();
+
+    // entry point for SRV_Channel to prod us to supply data to the
+    // AP_ESC_Telem library:
+    void update_telemetry();
+
+private:
+
+    AP_Int32 channel_mask;
+    AP_Int8 motor_poles;
+
+    static const uint8_t MAX_ESCS { 8 };
+
+    AP_HobbyWing_ESC *escs[MAX_ESCS];
+    uint8_t servo_channel[MAX_ESCS];
```

Comment:
These should be stored in the front-end surely - you are blowing up the amount of data required.

Reply:
This is the frontend?  We only create one of `AP_ESC_Telem_HobbyWing_Telem`, in the same way we only have one of the blheli manager thingy created in SRV_Channels.cpp

Reply:
@andyp1per perhaps you were suggesting we could do without `AP_ESC_Telemetry_HobbyWing` entirely and put these all of the members into `AP_ESC_Telemetry`?  I could definitely do that, but I would need to change the way the update_telemetry call gets triggered.  I emulated the way we've previously done things, by hooking the stuff in `SRV_Channels` - but I reckon there's a good argument to be made for putting the list-of-ESCs directly into `AP_ESC_Telemetry`.  Not sure how we'd handle the parameters at that point; hanging stuff off `SERVO_` is already pretty weird.

(p.s. I dislike AP_ESC_Telemetry_SITL - it keeps on overwriting my perfectly-good HobbyWing telemetry :-/ )

Reply:
IDK, I prefer not to break encapsulation - but there are also a number of parameters that we are duplicating everywhere. It would be good to at least have a consistent mechanism for that data. If its consistent and common then it could live in the front end